bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromBulbHandler.java

```diff
@@ -0,0 +1,314 @@
+                        }
+                        break;
+
+                    default:
+                }
+
+                if (sResp != null) {
+                    Type type = new TypeToken<HashMap<String, MyStromDeviceSpecificInfo>>() {
+                    }.getType();
+                    @Nullable
+                    Map<String, MyStromDeviceSpecificInfo> report = getGson().fromJson(sResp, type);
+                    if (report != null) {
+                        report.entrySet().stream().filter(e -> e.getKey().equals(getMac())).findFirst()
+                                .ifPresent(info -> updateDevice(info.getValue()));
+                    }
+                }
+            }
+        } catch (MyStromException e) {
+            logger.warn("Error while handling command {}", e.getMessage());
+        }
+    }
+
+    private @Nullable Map<String, MyStromDeviceSpecificInfo> getReport() {
+        try {
+            String returnContent = getDeviceSpecificInformation();
+            Type type = new TypeToken<HashMap<String, MyStromDeviceSpecificInfo>>() {
+            }.getType();
+            Map<String, MyStromDeviceSpecificInfo> report = getGson().fromJson(returnContent, type);
+            updateStatus(ThingStatus.ONLINE);
+            return report;
+        } catch (MyStromException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return null;
+        }
+    }
+
+    @Override
+    protected void pollDevice() {
+        @Nullable
+        Map<String, MyStromDeviceSpecificInfo> report = cache.getValue();
+        if (report != null) {
+            report.entrySet().stream().filter(e -> e.getKey().equals(getMac())).findFirst()
+                    .ifPresent(info -> updateDevice(info.getValue()));
+        }
+    }
+
+    private void updateDevice(@Nullable MyStromBulbResponse deviceInfo) {
+        if (deviceInfo != null) {
+            updateState(CHANNEL_SWITCH, deviceInfo.on ? OnOffType.ON : OnOffType.OFF);
+            @Nullable
+            HSBType color = null;
+            long numSemicolon = deviceInfo.color.chars().filter(c -> c == ';').count();
+            if (numSemicolon == 1 && deviceInfo.mode.equals(MONO)) {
+                String[] xy = deviceInfo.color.split(";");
+                lastColorTemperature = new QuantityType<>(xy[0]);
+                lastBrightness = PercentType.valueOf(xy[1]);
+                updateState(CHANNEL_COLOR_TEMPERATURE, lastColorTemperature);
+                updateState(CHANNEL_BRIGHTNESS, lastBrightness);
+                updateState(CHANNEL_MODE, StringType.valueOf(deviceInfo.mode));
+            } else if (numSemicolon == 2 && deviceInfo.mode.equals(HSV)) {
+                color = HSBType.valueOf(deviceInfo.color.replaceAll(";", ","));
+            } else if (!deviceInfo.color.equals("") && deviceInfo.mode.equals(RGB)) {
+                int r = Integer.parseInt(deviceInfo.color.substring(2, 4), 16);
```

Comment:
Maybe surround with `try-catch` for `IllegalArgumentException`, otherwise the scheduler might crash